src/components/PokemonTable.js

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import MUIDataTable from "mui-datatables";
+
+const renderTypes = types => types.join(" ");
+
+const shouldExcludePokemon = (pokemonTypes, filterTypes) => {
+  const result = filterTypes.every(val => pokemonTypes.includes(val));
+  return !result;
+};
+
+const getPokemonTypes = pokemon => [
+  ...new Set(pokemon.map(mon => [...mon.type, ...mon.weaknesses]).flat())
```

Comment:
originally you were creating two different arrays for valid types and weakness but if a pokemon cant be a type how could it be weak to it. with this you can avoid a whole array traversal and its more accurate because you should still be able to search for any valid type/weakness